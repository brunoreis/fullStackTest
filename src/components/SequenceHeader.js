/* responsibilities
show the header
show the config options
set the extra arguments
*/
import React from 'react';
import SequenceIntroAndConfigFields from './SequenceIntroAndConfigFields/SequenceIntroAndConfigFields';

const Buttons = ({
  pipeIsEven,
  pipeAccumulator,
  toggleAccumulator,
  toggleIsEven,
}) =>
  <div className="btn-group mb-2" role="group" aria-label="Third group">
    <button
      type="button"
      className={`btn btn-sm ${pipeAccumulator ? 'btn-secondary' : 'btn-default'}`}
      onClick={toggleAccumulator}
    >
      accumulator
    </button>
    <button
      type="button"
      className={`btn btn-sm ${pipeIsEven ? 'btn-secondary' : 'btn-default'}`}
      onClick={toggleIsEven}
    >
      isEven
    </button>
  </div>

export default ({
  extraArguments, // probably not needed anymore
  setExtraArguments,
  choosenSequence,
  next,
  sequenceStarted,
  toggleIsEven,
  toggleAccumulator,
  pipeIsEven,
  pipeAccumulator
}) =>
  <div>
    <h5 className="card-title d-flex  justify-content-around align-items-center">
      {
        /* <img src={logo} className="App-logo" alt="logo" /> */
      }
      <div>
        {choosenSequence.name}
      </div>
      <button
        disabled={extraArguments == null}
        className="btn btn-primary btn-sm"
        onClick = {next}
      >
        Next
      </button>
    </h5>
    <Buttons
      {...{
        pipeIsEven,
        pipeAccumulator,
        toggleAccumulator,
        toggleIsEven
      }}
    />
    {
      !sequenceStarted
      &&
      <SequenceIntroAndConfigFields
        {...{
          choosenSequence,
          setExtraArguments,
          extraArguments
        }}
      />
    }
  </div>
