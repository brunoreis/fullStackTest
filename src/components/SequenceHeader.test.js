// Link.react.test.js
import React from 'react';
import SequenceHeader from './SequenceHeader';
import renderer from 'react-test-renderer';

test('SequenceHeader shows the addAccumulator toggle button', () => {
  const component = renderer.create(
    <SequenceHeader
      choosenSequence = {{
        name: "Partial",
      }}
      setExtraArguments ={
        () => {console.log("setExtraArguments called")}
      }
      next ={
        () => {console.log("next called")}
      }
      sequenceStarted = {false}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
