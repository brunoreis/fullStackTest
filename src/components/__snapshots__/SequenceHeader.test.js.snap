// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SequenceHeader shows the addAccumulator toggle button 1`] = `
<div>
  <h5
    className="card-title d-flex  justify-content-around align-items-center"
  >
    <div>
      Partial
    </div>
    <button
      className="btn btn-primary btn-sm"
      disabled={true}
      onClick={[Function]}
    >
      Next
    </button>
  </h5>
  <div
    aria-label="Third group"
    className="btn-group mb-2"
    role="group"
  >
    <button
      className="btn btn-sm btn-default"
      type="button"
    >
      accumulator
    </button>
    <button
      className="btn btn-sm btn-default"
      type="button"
    >
      isEven
    </button>
  </div>
  <div>
    <p
      style={
        Object {
          "fontSize": "0.8em",
        }
      }
    />
  </div>
</div>
`;
